### PR TITLE
Jaeger bazel windows

### LINKED ISSUE #1101: Jaeger bazel Windows and Mac
**Is your feature request related to a problem?**
Jaeger bazel Windows and Mac require boost installed.

**Describe the solution you'd like**
ideally `bazel` should use `rules_foreign_cc` and add `boost` as dependency.
**Describe alternatives you've considered**
Which alternative solutions or features have you considered?

**Additional context**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1101 (issue) for Windows

## Changes

This PR uses `boost` and `openssl` from `vcpkg` for building `thrift`.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: ci/setup_thrift.ps1 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+$ErrorActionPreference = "Stop"
+trap { $host.SetShouldExit(1) }
+
+# TODO: consider moving this to .gitmodules
+git clone https://github.com/Microsoft/vcpkg.git
```
Comment (ThomsonTan) [inline ci/setup_thrift.ps1:5]: Checkout a stable release tag for reproducible build?

File: ci/setup_thrift.ps1 [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Push-Location -Path vcpkg
+$VCPKG_DIR=(Get-Item -Path ".\").FullName
+setx VCPKG_DIR (Get-Item -Path ".\").FullName
+./bootstrap-vcpkg.bat
+./vcpkg integrate install
```
Comment (ThomsonTan) [inline ci/setup_thrift.ps1:10]: Is this required for the build? This changes some global MSBuild settings which may interrupts existing vcpkg installation if any?

File: exporters/jaeger/BUILD [hunk 3]
```diff
@@ -1,36 +1,123 @@
...
+    "WITH_BOOST_FUNCTIONAL": "OFF",
+    "WITH_BOOST_SMART_PTR": "OFF",
+    "BUILD_SHARED_LIBS": "OFF",
+    "CMAKE_TOOLCHAIN_FILE": "$VCPKG_DIR/scripts/buildsystems/vcpkg.cmake",
+}
+
+THRIFT_DEPS_WIN = [
+]
+
+THRIFT_DEPS = THRIFT_DEPS_WIN + [
```
Comment (ThomsonTan) [inline exporters/jaeger/BUILD:62]: Are `THRIFT_DEPS` and `THRIFT_DEPS_WIN` expected to be extended to non-empty list later?

File: ci/setup_thrift.ps1 (hunk 1 above)
Comment (esigo) [reply ci/setup_thrift.ps1]: done.

File: exporters/jaeger/BUILD (hunk 3 above)
Comment (esigo) [reply exporters/jaeger/BUILD]: This was needed for Mac. I cleaned for now.

File: ci/setup_thrift.ps1 (hunk 2 above)
Comment (esigo) [reply ci/setup_thrift.ps1]: Thanks yeah integrate install is not needed. I cleaned.

**Review by ThomsonTan — APPROVED** [review]

**Review by lalitb — APPROVED** [review]

**Pull request merged**